govcd/lb_test.go

```diff
@@ -0,0 +1,356 @@
+// +build lb functional integration ALL
+// +build !skipLong
+
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+// Test_LB load balancer integration test
+// MANDATORY PREREQUISITE - manually enable load balancing for edge gateway with enabled acceleration
+// 1. Validates that all needed parameters are here
+// 2. Uploads or reuses media.photonOsOvaPath OVA image
+// 3. Creates RAW vApp and attaches vDC network to it
+// 4. Spawns two VMs with configuration script to server HTTP traffic
+// 5. Sets up load balancer
+// 6. Probes load balancer virtual server's external IP (edge gateway IP) for traffic
+// being server in 2 VMs
+// 7. Tears down
+func (vcd *TestVCD) Test_LB(check *C) {
+
+	// Validate prerequisites
+	validateTestLbPrerequisites(vcd, check)
+
+	// Get org and vdc
+	org, err := GetAdminOrgByName(vcd.client, vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	vdc, err := org.GetVdcByName(vcd.config.VCD.Vdc)
+	check.Assert(err, IsNil)
+
+	// Find catalog and check if we have
+	catalog, err := org.FindCatalog(vcd.config.VCD.Catalog.Name)
+	check.Assert(err, IsNil)
+
+	// Check if the image was already uploaded
+	ovaPath := vcd.config.Media.PhotonOsOvaPath
+	_, ovaFile := path.Split(ovaPath)
+	catalogItem, err := catalog.FindCatalogItem(ovaFile)
+	check.Assert(err, IsNil)
+
+	// If the image is not already uploaded - upload it
+	if catalogItem == (CatalogItem{}) || catalogItem.CatalogItem.Name == "" {
+		fmt.Printf("# OS image '%s' not found in catalog '%s. Uploading '%s'",
+			ovaFile, catalog.Catalog.Name, ovaPath)
+		// Upload OS image
+		uploadImage(ovaPath, ovaFile, catalog, vcd, check)
```

Comment:
Great test here. I'm only thinking whether it's this test's responsibility to upload an OVA. Maybe we should just skip it if there's no such OVA already uploaded by the test framework? @dataclouder your thoughts?

Reply:
Skipping test on missing prerequisites is a good practice.
We have many tests that depend on a catalog item. We can do the same here, if the item is missing.
We should keep the separate `photonOsOva` (not `Path`, as we don't want to upload during the test) in the configuration file, as the vApp template that we use for normal tests may be different. 
IMO, an empty `photonOsOva` or a missing catalog item as suggested by this field should just skip the test without errors.

Reply:
Ok. So I change it to `photonOsOva` and do not upload it. If it's not there - I skip.

Reply:
As per other comment I relly on catalogItem being Photon OS now. If it's no Photon OS - I skip the test.